cmd/crank/beta/xpkg/init.go

```diff
@@ -116,14 +133,132 @@ func (c *initCmd) Run(k *kong.Context, logger logging.Logger) error {
 		return errors.Wrapf(err, "failed to get repository's HEAD from %q", repoURL)
 	}
 
-	// TODO(phisco): replace placeholders for the name all around the
-	// 	repository? Maybe we can just agree on some markdown text in the
-	// 	repos to print to let the user know what to do next?
-
 	_, err = fmt.Fprintf(k.Stdout, "Initialized package %q in directory %q from %s (%s)\n", c.Name, c.Directory, getPrettyURL(logger, repoURL, ref), ref.Name().Short())
+	if err != nil {
+		return errors.Wrap(err, "failed to write to stdout")
+	}
+
+	err = handleNotes(k, logger, fs, c.Directory, c.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle NOTES.txt")
+	}
+
+	err = handleInitScript(k, logger, fs, c.Directory, c.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle initialization script")
+	}
+
 	return err
 }
 
+// handleNotes templates and prints the NOTES.txt file in the template
+// repository, if it exists.
+func handleNotes(k *kong.Context, logger logging.Logger, fs billy.Filesystem, dir, name string) error {
+	notesFile := dir + "/" + notes
+	f, err := fs.Stat(notesFile)
+	switch {
+	case os.IsNotExist(err):
+		// no NOTES.txt file, skip
+		logger.Debug("No NOTES.txt found, skipping")
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to stat notes file %s", notesFile)
+	case f.IsDir():
+		return errors.Errorf("%s is not a file", notesFile)
+	}
+
+	filler := nameFiller{Name: name}
+	nfi, err := fs.Open(notesFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open notes file %s", notesFile)
+	}
+	ntRaw, err := io.ReadAll(nfi)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read notes file %s", notesFile)
+	}
+	nt, err := template.New("notes").Parse(string(ntRaw))
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse notes file %s", notesFile)
+	}
+	err = nt.Execute(k.Stdout, filler)
+	if err != nil {
+		return errors.Wrapf(err, "failed to template notes file %s", notesFile)
+	}
```

Comment:
What are some reasons folks would want `NOTES.txt` to be a template?

I imagine it would explain how the extension template works, for example the `NOTES.txt` for https://github.com/crossplane/function-template-go would provide a little context on how to get started with function-template-go. I can't immediately think of a reason `NOTES.txt` would need to be templatized for that.

Reply:
Its just for making the NOTES.txt a bit more personalised based on the name you chose. 
So instead of having:
```
Replace function-template-go with your function's name in go.mod, package/crossplane.yaml, and any Go imports
```
You can have:
```
Replace function-template-go with somename in go.mod, package/crossplane.yaml, and any Go imports
```

If there is a NOTES.txt with no templating in it, it will still get printed out.


TBH its just cosmetic and I dont feel to strongly about it. I can remove it for keeping the feature simpler.

Reply:
I lean toward removing it to keep it simpler. I don't feel very strongly about it either though. 😄

Reply:
Ill remove it for simplicity. It will be easier to add it later then remove.

Reply:
i'm seeing a raw `{{ .Name }}` in the output when testing now - is this because you removed the templating ability, but the NOTES.file hasn't been updated to remove the templated var?

```
❯ go run cmd/crank/main.go beta xpkg init func https://github.com/lsviben/function-template-go -d init-test
Initialized package "func" in directory "/Users/jared/dev/go/src/github.com/crossplane/crossplane/init-test" from https://github.com/lsviben/function-template-go/tree/a5e76778657cd80bfe8100e77608f0577f75e291 (main)

To get started:

1. Replace `function-template-go` with {{ .Name }} in `go.mod`,
   `package/crossplane.yaml`, and any Go imports
...
```

Reply:
I just didnt update the my test function repo yet. The real NOTES.txt and init.sh that will go into the official repo will be proper.